web/modules/custom/dpl_po/src/Commands/DplPoCommands.php

```diff
@@ -0,0 +1,432 @@
+      if (!$uri = $this->fileSystem->tempnam('temporary://', 'po_')) {
+        $this->io()->error($this->t('Could not create temp PO file.'));
+        return;
+      }
+
+      $header = new PoHeader($langcode);
+      $header->setProjectName($this->configFactory->get('system.site')->get('name'));
+      $header->setLanguageName($langcode);
+
+      $writer = new PoStreamWriter();
+      $writer->setURI($uri);
+      $writer->setHeader($header);
+
+      $writer->open();
+      foreach ($items as $item) {
+        $writer->writeItem($item);
+      }
+      $writer->close();
+
+      if (!is_string($uri) || !$file = $this->fileSystem->realpath($uri)) {
+        $this->io()->error($this->t('Could not locate temp PO file.'));
+        return;
+      }
+
+      try {
+        $destination = $this->moveFile(new \SplFileInfo($file));
+      }
+      catch (\Exception $exception) {
+        $this->io()->error($this->t('Could not create PO file: @file (exportConfigPoFile)', ['@file' => $uri]));
+        return;
+      }
+
+      $this->io()->success($this->t('File created on: @destination', ['@destination' => $destination]));
+    }
+  }
+
+  /**
+   * Validate if source file exists and is readable.
+   *
+   * @throws \Exception
+   */
+  protected function validateSource(): void {
+    $source = $this->getSource();
+
+    if (!$source || !is_file($source)) {
+      throw new \Exception('Invalid source file: ' . $source);
+    }
+
+    if (!is_readable($source)) {
+      throw new \Exception('Unreadable source file: ' . $source);
+    }
+  }
+
+  /**
+   * Validate if destination directory exists and is writable.
+   *
+   * @throws \Exception
+   */
+  protected function validateDestination(): void {
+    if (!$destination = $this->getDestination()) {
+      throw new \Exception('Destination was not defined.');
+    }
+
+    // Check for writable destination.
+    $destination_dir = $this->fileSystem->dirname($destination);
+    if (!is_writable($destination_dir)) {
+      throw new \Exception('Destination dir is not writable: ' . $destination_dir);
+    }
+  }
+
+  /**
+   * Create a file object.
+   */
+  protected function createFile(string $path): \stdClass {
+    $file = new \stdClass();
+    $file->filename = $this->fileSystem->basename($path);
+    $file->uri = $path;
+    $file->langcode = $this->getLanguageCode();
+    $file->timestamp = filemtime($path);
+
+    return $file;
+  }
+
+  /**
+   * Move a file to the destination.
+   */
+  protected function moveFile(\SplFileInfo $file): ?string {
+    if ($destination = $this->getDestination()) {
+      rename($file->getRealPath(), $destination);
+    }
+
+    return $destination;
+  }
+
+  /**
+   * Extract translations into a file.
+   */
+  protected function extractTranslationsIntoFile(string $pattern, string $mode = 'include'): \SplFileInfo {
+    $source = $this->getSource();
+    if (!$source = $this->getSource()) {
+      throw new \Exception('Source is not defined.');
+    }
```

Comment:
Please consider if this check should this check be replaced by `validateSource()`.